### PR TITLE
Add the unignitable trait

### PR DESCRIPTION
**Pull request by TiviPlus**
Backend 4 robots

### CODE REVIEW

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -118,6 +118,8 @@
 #define TRAIT_SEE_IN_DARK "see_in_dark" //Able to see in dark
 #define TRAIT_MUTED "muted" //target is mute and can't speak
 #define TRAIT_TURRET_HIDDEN "turret_hidden" //target gets passed over by turrets choosing a victim
+///Prevent mob from being ignited due to IgniteMob()
+#define TRAIT_UNIGNITABLE "unignitable"
```
Comment (psykzz) [inline code/__DEFINES/traits.dm:122]: Inflammable?

Comment (BraveMole) [reply code/__DEFINES/traits.dm:122]: ininflammable but yes

**Review by BraveMole — CHANGES_REQUESTED** [review]
So i was half joking.

Inflammable means can be ingited

Ininflammable means cannot be ignited, but in french.

Non-Flammabe, or fire resitant, would be better in english

**Comment by TiviPlus** [comment]
Fire resistant is already taken on TG for being semi-fireproof to some things(not this) and i'd rather avoid in case theres a conlfict

iningnitable isnt a word in english

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -118,6 +118,8 @@
 #define TRAIT_SEE_IN_DARK "see_in_dark" //Able to see in dark
 #define TRAIT_MUTED "muted" //target is mute and can't speak
 #define TRAIT_TURRET_HIDDEN "turret_hidden" //target gets passed over by turrets choosing a victim
+///Prevent mob from being ignited due to IgniteMob()
+#define TRAIT_INFLAMMABLE "inflammable"
```
Comment (BraveMole) [inline code/__DEFINES/traits.dm:122]:
```suggestion
#define TRAIT_NON_FLAMMABLE "non-flammable"
```

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -130,6 +130,8 @@
 /mob/living/proc/IgniteMob()
 	if(status_flags & GODMODE) //Invulnerable mobs don't get ignited
 		return FALSE
+	if(HAS_TRAIT(src, TRAIT_INFLAMMABLE))
```
Comment (BraveMole) [inline code/modules/mob/living/living_defense.dm:133]:
```suggestion
	if(HAS_TRAIT(src, TRAIT_NON_FLAMMABLE))
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Pull request merged**